src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -558,31 +558,34 @@ public void MaxRequestContentBufferSize_Set_ThrowsPlatformNotSupportedException(
             var cookie2 = new KeyValuePair<string, string>(".AspNetCore.Session", "RAExEmXpoCbueP_QYM");
             var cookie3 = new KeyValuePair<string, string>("name", "value");
 
-            string url = string.Format(
-                "http://httpbin.org/cookies/set?{0}={1}&{2}={3}&{4}={5}",
-                cookie1.Key,
-                cookie1.Value,
-                cookie2.Key,
-                cookie2.Value,
-                cookie3.Key,
-                cookie3.Value);
-
-            var handler = new HttpClientHandler()
+            await CreateServerAsync(async (handler, client, server, url) =>
             {
-                AllowAutoRedirect = false
-            };
+                Task<HttpResponseMessage> getResponse = client.GetAsync(url);
 
-            using (var client = new HttpClient(handler))
-            using (HttpResponseMessage response = await client.GetAsync(url))
-            {
-                Assert.Equal(HttpStatusCode.Redirect, response.StatusCode);
-                CookieCollection cookies = handler.CookieContainer.GetCookies(new Uri(url));
+                await AcceptSocketAsync(server, async (s, stream, reader, writer) =>
+                {
+                    while (!string.IsNullOrEmpty(reader.ReadLine())) ;
+                    await writer.WriteAsync(
+                        $"HTTP/1.1 200 OK\r\n" +
```

Comment:
I think if we want to test the low level HTTP stacks, then it's fine to try out a response with no Date: header in a set of tests that explicitly test for that. But as a best-practice, we should always have the Date: header in our responses. It also makes it easier to diagnose since we have a timestamp.